### PR TITLE
feat(pipelines): Add ability to define fileSystemLocations for a CodePipeline

### LINKED ISSUE #24495: aws-codebuild: Add file systems to BuildEnvironment
### Describe the feature

Introduce the ability to define file systems within the BuildEnvironment for aws-codebuild.

### Use Case

Currently you're able to add file systems to a CodeBuild project using the console or by defining fileSystemLocations when initializing a CodeBuild Project using the cdk. However, in cases where the creation of CodeBuild projects are abstracted away, this is not possible. Such case is pipelines.CodePipeline that allows setting codeBuildDefaults' buildEnvironment, which allows for setting of buildImage, certificate, computeType, environmentVariables and privileged parameters, but not the fileSystemLocations. This feature would help greatly with mounting EFS drives for CodePipelines, as I am currently not aware of a method to do this simply with the CDK.

### Proposed Solution

Add fileSystemLocations to BuildEnvironment struct and introduce way to set CodeBuild project's fileSystemLocations via the environment props.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.67.0

### Environment details (OS name and version, etc.)

Windows 10

### PR DESCRIPTION
**Pull request by lassiraa**
This adds the ability to define fileSystemLocations for CodePipeline.  Currently, there is not a simple way (or any way that I know of) to do this. This will come in handy when e.g. attaching EFS filesystems to CodePipelines.

Closes #24495.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by lassiraa** [comment]
Clarification Request
Hi, currently I am having issues with my integration test. The stack deployment fails with error message: 'CREATE_FAILED (The following resource(s) failed to create: [Pipeline9850B417]. ): Internal Failure' and I am currently a bit stuck as to how to proceed. Could I have someone look at this and at least point me into the right direction?
Thanks already!

**Comment by lassiraa** [comment]
Alright, it seems that the synth input was likely the cause of my earlier issues. The integration tests pass now and I was able to run my own integration test as well. Hopefully everything looks good here now!

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

File: packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline.ts [hunk 1]
```diff
@@ -307,6 +307,17 @@ export interface CodeBuildOptions {
    * @default Duration.hours(1)
    */
   readonly timeout?: Duration;
+
+  /**
+   * ProjectFileSystemLocation objects for CodeBuild build projects.
+   *
+   * A ProjectFileSystemLocation object specifies the identifier, location, mountOptions, mountPoint,
+   * and type of a file system created using Amazon Elastic File System.
+   * Requires a vpc to be set and privileged to be set to true.
```
Comment (otaviomacedo) [inline packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline.ts:316]: You should validate whether these requirements were met, then.

**Review by otaviomacedo — CHANGES_REQUESTED** [review]

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**